pdr_backend/pdr_dashboard/util/data.py

```diff
@@ -126,3 +131,95 @@ def find_with_key_value(
             return obj
 
     return None
+
+
+def get_feeds_stat_with_contract(
+    contract: str, feed_stats: List[Dict[str, Any]]
+) -> Dict[str, Union[float, int]]:
+    result = find_with_key_value(feed_stats, "contract", contract)
+
+    if result:
+        return {
+            "avg_accuracy": float(result["avg_accuracy"]),
+            "avg_stake_per_epoch_(OCEAN)": float(result["avg_stake"]),
+            "volume_(OCEAN)": float(result["volume"]),
+        }
+
+    return {
+        "avg_accuracy": 0,
+        "avg_stake_per_epoch_(OCEAN)": 0,
+        "volume_(OCEAN)": 0,
+    }
+
+
+def get_feeds_subscription_stat_with_contract(
+    contract: str, feed_subcription_stats: List[Dict[str, Any]]
+) -> Dict[str, Union[float, int, str]]:
+    result = find_with_key_value(feed_subcription_stats, "contract", contract)
+
+    if result:
+        sales_str = f"{result['sales']}"
+
+        df_buy_count_str = (
+            f"{result['df_buy_count']}-DF" if result["df_buy_count"] > 0 else ""
+        )
+        ws_buy_count_str = (
+            f"{result['ws_buy_count']}-WS" if result["ws_buy_count"] > 0 else ""
+        )
+
+        if df_buy_count_str or ws_buy_count_str:
```

Comment:
or alternately build the result itself. It could also improve this part right here modify result["sales"] if needed